src/overlays/actors/ovl_En_Ruppecrow/z_en_ruppecrow.c

```diff
@@ -81,62 +112,558 @@ static DamageTable D_80BE39EC = {
+        this->rupees[rupeeIndex] = rupee;
+        this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+        this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+        rupee = this->rupees[rupeeIndex];
+        rupee->unk152 = 60;
+        this->rupees[rupeeIndex]->actor.flags |= 0x10;
+    } else {
+        rupee =
+            (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, this->actor.world.pos.x + xOffset,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0x0, 0x0, 0x0, ITEM00_RUPEE_GREEN);
+        this->rupees[rupeeIndex] = rupee;
+        this->rupees[rupeeIndex]->actor.gravity = -5.0f;
+        this->rupees[rupeeIndex]->actor.velocity.y = 0.0f;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_RUPY_FALL);
+        rupee = this->rupees[rupeeIndex];
+        rupee->unk152 = 60;
+        this->rupees[rupeeIndex]->actor.flags |= 0x10;
+    }
+
+    this->rupeeIndex++;
+}
+
+void EnRuppecrow_UpdatePosition(EnRuppecrow* this, GlobalContext* globalCtx) {
+    Vec3s nextPointDirection;
+
+    EnRuppecrow_GetPointDirection(this->path, this->currentPoint, &this->actor.world, &nextPointDirection);
+    if (this->actor.bgCheckFlags & 0x8) {
+        nextPointDirection.y = this->actor.wallYaw;
+    }
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, nextPointDirection.y, 0x4, 0x3E8, 0x1);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Math_SmoothStepToS(&this->actor.world.rot.x, -nextPointDirection.x, 0x4, 0x3E8, 0x1);
+
+    if (this->isGoingCounterClockwise & 1) {
+        if (EnRuppecrow_ReachedPointCounterClockwise(this, this->path, this->currentPoint)) {
+            if (this->currentPoint <= 0) {
+                this->currentPoint = this->path->count - 1;
+            } else {
+                this->currentPoint--;
+            }
+
+            if (this->actionFunc == EnRuppecrow_FlyWhileDroppingRupees &&
+                (!EnRuppecrow_CanSpawnBlueRupees(globalCtx) || (this->currentPoint % -2) == 0)) {
+                EnRuppecrow_SpawnRupee(this, globalCtx);
+            }
+        }
+    } else if (EnRuppecrow_ReachedPointClockwise(this, this->path, this->currentPoint)) {
+        if (this->currentPoint >= this->path->count - 1) {
+            this->currentPoint = 0;
+        } else {
+            this->currentPoint++;
+        }
+
+        if (this->actionFunc == EnRuppecrow_FlyWhileDroppingRupees &&
+            (!EnRuppecrow_CanSpawnBlueRupees(globalCtx) || (this->currentPoint % -2) == 0)) {
+            EnRuppecrow_SpawnRupee(this, globalCtx);
+        }
+    }
+}
+
+s32 EnRuppecrow_CheckPlayedMatchingSong(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (globalCtx->msgCtx.unk1202A == 0x3) {
+        switch (player->transformation) {
+            case PLAYER_FORM_DEKU:
+                if (globalCtx->msgCtx.unk1202E == OCARINA_SONG_SONATA) {
```

Comment:
How did you knew `globalCtx->msgCtx.unk1202E` is an ocarina song?

Reply:
It's like a 90% guess based on:

- There is an ocarina song interaction with the guay, a successful song check will cause him to spawn rupees
- The cases match up (each transformation has to play their own temple song, but human link can play any of them)
- The enum values match up with the songs

It's not 100%, but based on the context and gameplay it seemed likely to me that that's what they are.